tests/fixtures/optimizer/optimizer.sql

```diff
@@ -154,3 +154,44 @@ SELECT
 FROM VALUES
   (1, 'test'),
   (2, 'test2') AS `tab`(`cola`, `colb`);
+
+WITH m AS (
+  SELECT a, b FROM (VALUES (1, 2)) AS a1(a, b)
+), n AS (
+  SELECT a, b FROM m WHERE m.a = 1
+), o AS (
+  SELECT a, b FROM m WHERE m.a = 2
+) SELECT
+    n.a,
+    n.b,
+    o.b
+FROM n
+FULL OUTER JOIN o ON n.a = o.a
+CROSS JOIN n AS n2
+WHERE o.b > 0 AND n.a = n2.a;
```

Comment:
This is a strange example since having the `n.a` = `o.a` in the where clause makes the query return nothing since n is null.

Reply:
didn't you give me this example? i just added some stuff to it